protocol/msgpack/unaggregated_iterator.go

```diff
@@ -219,37 +222,62 @@ func (it *unaggregatedIterator) decodeGauge() {
 	it.skip(numActualFields - numExpectedFields)
 }
 
-func (it *unaggregatedIterator) decodeVersionedPolicies() {
+func (it *unaggregatedIterator) decodePoliciesList() {
 	numActualFields := it.decodeNumObjectFields()
-	versionedPoliciesType := it.decodeObjectType()
-	numExpectedFields, numActualFields, ok := it.checkNumFieldsForTypeWithActual(
-		versionedPoliciesType,
+	policiesListType := it.decodeObjectType()
+	numExpectedFields, ok := it.checkExpectedNumFieldsForType(
+		policiesListType,
 		numActualFields,
 	)
 	if !ok {
 		return
 	}
-	version := it.decodeVersion()
-	cutover := it.decodeTime()
-	switch versionedPoliciesType {
-	case defaultVersionedPoliciesType:
-		it.versionedPolicies = policy.DefaultVersionedPolicies(version, cutover)
-		it.skip(numActualFields - numExpectedFields)
-	case customVersionedPoliciesType:
-		numPolicies := it.decodeArrayLen()
-		if cap(it.policies) < numPolicies {
-			it.policies = make([]policy.Policy, 0, numPolicies)
+	switch policiesListType {
+	case defaultPoliciesListType:
+		it.policiesList = policy.DefaultPoliciesList
+	case customPoliciesListType:
+		numStagedPolicies := it.decodeArrayLen()
+		if cap(it.cachedPoliciesList) < numStagedPolicies {
+			it.cachedPoliciesList = make(policy.PoliciesList, 0, numStagedPolicies)
 		} else {
-			it.policies = it.policies[:0]
+			it.cachedPoliciesList = it.cachedPoliciesList[:0]
+		}
+		if len(it.cachedPolicies) < numStagedPolicies {
+			it.cachedPolicies = make([][]policy.Policy, numStagedPolicies)
```

Comment:
I see, and `policy.Policy` is a value type that has no fields that are pointers yeah?